DOCKER.md

```diff
@@ -0,0 +1,53 @@
+<!--
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# accumulo-proxy-docker
+
+A temporary guide on how to run this up in Docker.
+
+## Build the image using
+Invoke the docker build command to create a container image.
+```commandline
+docker build -t accumulo-proxy:latest .
+```
+
+## Default Configuration
+By default, the container image expects the following to be true:
+1. Your accumulo instance is named "myinstance"
+2. Your zookeeper is available (and reachable from the container) at localhost:2181
+
+## Custom proxy.properties
+If you wish to create advanced proxy.properties configuration changes, you should look to volume mount these in when you invoke the `docker run` command, an example is:
+```commandline
+docker run --rm -d -p 42424:42424 -v /path/to/proxy.properties:/opt/accumulo-proxy/conf/proxy.properties --add-host"FQDN:IP" --name accumulo-proxy accumulo-proxy:latest
+```
+
+## Accumulo Instance Configuration
+In order for Thrift to communicate with the Accumulo instance you will need to provide the container with the FQDN and IP of the Accumulo instance and its relevant servers (e.g. tservers) to allow it to resolve the required DNS.
+
+If you are running an Accumulo instance with more than one tserver you should add each tserver's entry with a new `--add-host "FQDN:IP"` entry.
```

Comment:
Instead of adding an entry for each tserver, may be able to use the the docker run option `--network="host"`.

Reply:
@keith-turner so that's true in certain environments, this is where I think we have to make a decision about what to support/not support. 

In standard linux generally the docker configuration is fairly close to bare metal.

If you use Windows or in my case Mac generally you run Docker 'technically' inside a VM, therefore when you attach the host network it's still not technically the 'host (physical machine)' network. 

So whilst adding `--network="host"` into the instructions it'll only actually work for a subset of environments, adding the `--add-host` would work for a wider number and also allows for some isolation of resources (maybe a step too far for dev).

Also this all changes again if we look towards any other orchestration tooling such as docker-compose or Kubernetes as both of those have different working configurations which then need different tuning to work.

---

Given this is starting to creep into environment specific, I might comment on both options in the DOCKER.md and let the caller decide how they want to start their environment, but I'll try and give some hints as to what people may wish to consider.

For now, I don't think many of us are intending Docker to be a simple non-technically orientated mechanism to setup so we can probably get away with giving options for now.

Reply:
Could have a sentence like the following.  

```
Instead of having an `--add-host` option for each Accumulo server, a single `--network="host"` may work in some environments.
```

Reply:
Take a look at the latest update, I decided to re-write the whole section covering networking to try and give a bit more clarity around options but I also tried to avoid re-writing the docker documentation, hopefully this strikes a balance.

Reply:
I'm going to raise an issue to capture the proxy.properties overrides from the command line, I would like to discuss potential options before implementing these.

Reply:
> I decided to re-write the whole section covering networking to try and give a bit more clarity around options but I also tried to avoid re-writing the docker documentation, hopefully this strikes a balance.

The documentation is really nice.  IMO a good goal for this is a minimum amount of docker documentation to help someone get started quickly.  I think what you wrote  strikes a nice balance.